src/commands/test/lib/state-checker.ts

```diff
@@ -28,12 +28,29 @@ export class StateChecker {
 
   public async checkUntilCompleted(timeoutSec: number = null): Promise<number> {
     let exitCode = 0;
+    let errorCount = 0;
+    const maxErrors = 60;
+    const errorRetryWait = 2;
     const startTime = process.hrtime();
     if (this.isInternalStreamingOutput) {
       this.streamingOutput.start();
     }
     while (true) {
-      const state = await out.progress("Checking status...", this.getTestRunState(this.client, this.testRunId));
+      let state;
+      try {
+        state = await out.progress("Checking status...", this.getTestRunState(this.client, this.testRunId));
+      } catch (error) {
+        errorCount++;
+        if (errorCount >= maxErrors) {
+          throw error;
+        }
+
+        await out.progress("Status check failed, retrying...", this.delay(1000 * errorRetryWait));
+        continue;
```

Comment:
This shortcuts the existing timeout behavior. Please make sure the method still exits after `timeoutSec` seconds.